RankPanda/Rank.py

```
@@ -255,39 +264,9 @@ def AddBasicCommand(self, commandName, number, length, name):
-        elif ((commandName == "BackMarch") or (commandName == "BM")):
-            newCommand = Commands.BackMarch(length, beginLocation)
-        elif ((commandName == "RightSlide") or (commandName == "RS")):
-            newCommand = Commands.RightSlide(length, beginLocation)
-        elif ((commandName == "LeftSlide") or (commandName == "LS")):
-            newCommand = Commands.LeftSlide(length, beginLocation)
-        elif (commandName == "GTCCW0"):
-            newCommand = Commands.GTCCW0(length, beginLocation)
-        elif (commandName == "GTCW0"):
-            newCommand = Commands.GTCW0(length, beginLocation)
-        elif (commandName == "GTCCW1"):
-            newCommand = Commands.GTCCW1(length, beginLocation)
-        elif (commandName == "GTCW1"):
-            newCommand = Commands.GTCW1(length, beginLocation)
-        elif (commandName == "PWCCW"):
-            newCommand = Commands.PWCCW(length, beginLocation)
-        elif (commandName == "PWCW"):
-            newCommand = Commands.PWCW(length, beginLocation)
-        elif ((commandName == "Expand0") or (commandName == "Exp0")):
-            newCommand = Commands.Expand0(length, beginLocation)
-        elif ((commandName == "Expand1") or (commandName == "Exp1")):
-            newCommand = Commands.Expand1(length, beginLocation)
-        elif ((commandName == "Condense0") or (commandName == "Cond0")):
-            newCommand = Commands.Condense0(length, beginLocation)
-        elif ((commandName == "Condense1") or (commandName == "Cond1")):
-            newCommand = Commands.Condense1(length, beginLocation)
-        elif ((commandName == "Flatten") or (commandName == "Flat")):
-            newCommand = Commands.Flatten(length, beginLocation)
+        
+		#(Brady) simplified this by adding a dictionary to hash names to the function
+        newCommand = commandHash[commandName](length, beginLocation) #May want to catch a KeyError in case of typo or something...

```

Please catch the key error and fail with a useful error message like "Unknown command: foo"